Source/Kernel/Grains/Observation/States/ReplayEvaluator.cs

```diff
@@ -24,6 +24,11 @@ public class ReplayEvaluator(
     /// <inheritdoc/>
     public async Task<bool> Evaluate(ReplayEvaluationContext context)
     {
+        if (!context.State.IsReplayable)
```

Comment:
Encapsulate this in a method called `CanReplay` and combine this check with the `NeedsToReplay`